stcgal/protocols.py

```diff
@@ -260,14 +260,18 @@ def connect(self, autoreset=False):
         self.ser.flushInput()
 
         if autoreset:
-            print("Cycling power: ", end="")
-            sys.stdout.flush()
-            self.ser.setDTR(True)
-            time.sleep(0.5)
-            self.ser.setDTR(False)
-            print("done")
-            print("Waiting for MCU: ", end="")
-            sys.stdout.flush()
+            if not resetcmd:
+                print("Cycling power: ", end="")
+                sys.stdout.flush()
+                self.ser.setDTR(True)
+                time.sleep(0.5)
+                self.ser.setDTR(False)
+                print("done")
+                print("Waiting for MCU: ", end="")
+                sys.stdout.flush()
+            else:
+                print("Cycling power via shell cmd: " + resetcmd)
+                os.system(resetcmd)
```

Comment:
The connect() is starting to become a bit unwieldy. Can you extract a method for the autoreset functionality in a separate commit?